crates/sui-node/src/main.rs

```diff
@@ -59,63 +52,10 @@ async fn main() -> Result<()> {
-                // many statistics are cached and only updated when the epoch is advanced.
-                epoch::advance().unwrap();
-
-                // NOTE: The below code does not return values when a malloc-based profiler like Bytehound
-                // is used.  Bytehound does not implement the stat APIs needed.
-                let allocated = stats::allocated::read().unwrap() / (1024 * 1024);
-                let resident = stats::resident::read().unwrap() / (1024 * 1024);
-                info!(
-                    "Jemalloc: {} MB allocated / {} MB resident",
-                    allocated, resident
-                );
-
-                // TODO: split this out into mysten-infra so everyone can pick it up
-                // The reason why we use manual code to dump out profiles is because the automatic JEPROF
-                // options dump out too often.  We really just want profiles when the retained memory
-                // keeps growing, as we want to know why.
-                // Setting the timestamp and memory size in the filename helps us pick the profiles
-                // to use when doing analysis.  Default JEPROF profiles just have a counter which is not
-                // helpful.  This helps correlate to time and total memory consumed.
-                //
-                // NOTE: One needs to set MALLOC_CONF to `prof:true` for the below to work
-                if (allocated as f64 / last_allocated_mb as f64) > MEMORY_INCREASE_PROFILING_RATIO {
-                    info!("Significant memory increase registered, dumping profile: new = {}, old = {}",
-                          allocated, last_allocated_mb);
-                    last_allocated_mb = allocated;
-
-                    // Formulate profiling filename based on ISO8601 timestamp and number of MBs
-                    let dt = chrono::offset::Local::now();
-                    let dt_str = dt.to_rfc3339_opts(chrono::SecondsFormat::Secs, true);
-                    let dump_name = format!("jeprof.{}.{}MB.heap", dt_str, allocated);
-
-                    // Trigger profiling dump
-                    let dump_name_cstr = CString::new(dump_name).expect("Cannot create dump name");
-                    unsafe {
-                        if jemalloc_ctl::raw::write(PROF_DUMP, dump_name_cstr.as_ptr()).is_err() {
-                            debug!("Cannot dump memory profile, is _RJEM_MALLOC_CONF set to prof:true?");
-                        }
-                    }
-                }
-                std::thread::sleep(Duration::from_secs(MEMORY_PROFILING_INTERVAL_SECS));
-            }
-        });
-    }
+    // Spins up a thread to check memory usage every minute, and dump out stack traces/profiles
+    // if it has moved up or down more than 15%.  Also allow configuration of dump directory.
+    let profile_dump_dir = std::env::var("SUI_MEM_PROFILE_DIR").unwrap_or_default();
+    ProfilerRunner::new(60, 15, &profile_dump_dir).spawn();
```

Comment:
how is the 15% calculated? compared to the last time checks were performed or percentage of total mem?

Reply:
answering my own q: it's [relative change](https://github.com/velvia/ying-profiler/blob/main/src/utils.rs#L66)

Reply:
Yup